### PR TITLE
chore(TS): Remove ts-nocheck from pattern and add types

### PR DESCRIPTION
**Pull request by asturur**
## Motivation
Finish TS migration for this file

## Changes
Removed the setOptions method for Pattern.

### CODE REVIEW

**Comment by asturur** [comment]
Actions are not starting for me today, i wonder what is going on.

**Comment by asturur** [comment]
image

File: src/pattern.class.ts [hunk 1]
```diff
@@ -141,7 +137,7 @@ export class Pattern {
           this.source.naturalWidth === 0 ||
           this.source.naturalHeight === 0))
     ) {
-      return '';
+      return null;
```
Comment (ShaMan123) [inline src/pattern.class.ts:140]: doesn't this cause a problem in canvas?
I understand why you prefer this
I would look into usage and do `|| ''`

**Review by ShaMan123 — APPROVED** [review]

File: src/pattern.class.ts (hunk 1 above)
Comment (asturur) [reply src/pattern.class.ts:140]: i tried, if you set up a non valid color for canvas, you still get black.
so null or empty string it doesn't change.
empty string is as non valid as null for a fillStyle/strokeStyle.

createPattern api from context can also return null if pattern creation fails.

**Pull request merged**